lib/liquid/template_factory.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+module Liquid
+  class TemplateFactory
```

Comment:
I think the only issue with this approach is you need to build a whole new `Template` which means needing to register all the basic tags again. Would an approach similar to how context creates a restricted version of itself with `context.new_isolated_subcontext` be used here with a `template.new_isolated_subtemplate`

Reply:
After seeing more of the plan to implement this, I think this can likely stay similar.